### PR TITLE
parse `rule.meta.mbc` on load

### LINKED ISSUE #526: parse MBC technique into fields for the JSON output format
> structuring it, like the rest of the JSON output e.g. {"ID":"B0001.008","tactic":"Anti-Behavioral Analysis","technique":"Debugger Detection","subtechnique":"IsDebuggerPresent"}, following https://raw.githubusercontent.com/MBCProject/mbc-markdown/master/yfaq/mbc_matrix_with_ids.svg

> parsing the string for the json output is a good idea - it reduces the number of times that programmatic users of capa have to write one-off parsers for the MBC information.

from https://github.com/fireeye/capa-rules/issues/316

### PR DESCRIPTION
**Pull request by mr-tz**
### Description

place to discuss a proposed breaking change of the rule meta section

if we agree on this approach, do this for att&ck as well

**edit: as you can see this currently causes issues with the rule formatter;
alternatively, do this only when emitting the json**

closes #526

### Documentation

- [ ] I have updated the [CHANGELOG.md](/CHANGELOG.md), this is required for:
  - Bug fixes (non-breaking change which fixes an issue)
  - New features (non-breaking change which adds functionality)
  - Breaking changes (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update
  - [ ] I have made the corresponding changes to the documentation

### Tests

- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] No new tests needed

### CODE REVIEW

File: capa/rules.py [hunk 1]
```diff
@@ -277,6 +277,42 @@ def parse_description(s, value_type, description=None):
...
+    else:
+        behavior, _, id = rest.rpartition(" ")
+        method = ""
+    return {
+        "id": id.lstrip("[").rstrip("]"),
+        "objective": objective,
+        "behavior": behavior,
+        "method": method,
+        # TODO "micro-behavior": "",
+    }
```
Comment (williballenthin) [inline capa/rules.py:313]: im concerned here that we expect exactly two or three components and never any further subscope. we should parse and expose all the components, even if there are seven. so perhaps something like:

```suggestion
    path = mbc.split("::")
    if len(path) > 0:
        last = path.pop()
        last, _, id = last.partition(" ")
        id = id.lstrip("[").rstrip("]")
        path.append(last)
    
    if len(path) > 0:
        objective = path[0]
    if len(path) > 1:
        behavior = path[1]
    if len(path) > 2:
        method = path[2]

    return {
        "path": path,
        "id": id or "",
        "objective": objective or "",
        "behavior": behavior or "",
        "method": method or "",
    }
```

**Comment by mr-tz** [comment]
superseded by #576

File: capa/rules.py [hunk 2]
```diff
@@ -277,6 +277,42 @@ def parse_description(s, value_type, description=None):
...
+def parse_meta(meta):
+    """
+    further process meta items such as MBC
+    """
+    mbcs = meta.get("mbc", [])
+    if not isinstance(mbcs, list):
+        raise InvalidRule("MBC mapping must be a list")
+
+    if mbcs:
+        meta["mbc"] = [parse_canonical_mbc(mbc) for mbc in mbcs]
```
Comment (williballenthin) [inline capa/rules.py:289]:
```suggestion
    meta["mbc"] = [parse_canonical_mbc(mbc) for mbc in mbcs]
```

Comment (williballenthin) [reply capa/rules.py:289]: this way the key is always present and simplifies client code (don't have to check for presence)

**Review by williballenthin — CHANGES_REQUESTED** [review]
concerns about not being able to parse out all of the MBC path

**Pull request closed without merging**